modules/non_summary_pipeline.py

```diff
@@ -0,0 +1,501 @@
+    """Parses the page from a HAR object."""
+
+    if not har:
+        return None
+
+    page = har.get("log").get("pages")[0]
+    url = page.get("_URL")
+
+    metadata = get_metadata(har)
+    if metadata:
+        # The page URL from metadata is more accurate.
+        # See https://github.com/HTTPArchive/data-pipeline/issues/48
+        url = metadata.get("tested_url", url)
+
+    try:
+        payload_json = to_json(page)
+    except Exception:
+        logging.warning(
+            'Skipping pages payload for "%s": unable to stringify as JSON.' % url
+        )
+        return None
+
+    payload_size = len(payload_json)
+    if payload_size > MAX_CONTENT_SIZE:
+        logging.warning(
+            'Skipping pages payload for "%s": payload size (%s) exceeds the maximum content size of %s bytes.'
+            % (url, payload_size, MAX_CONTENT_SIZE)
+        )
+        return None
+
+    return [
+        {
+            "url": url,
+            "payload": payload_json,
+            "date": har["date"],
+            "client": har["client"],
+            "metadata": metadata,
+        }
+    ]
+
+
+def get_page_url(har):
+    """Parses the page URL from a HAR object."""
+
+    page = get_page(har)
+
+    if not page:
+        logging.warning("Unable to get URL from page (see preceding warning).")
+        return None
+
+    return page[0].get("url")
+
+
+def get_metadata(har):
+    page = har.get("log").get("pages")[0]
+    metadata = page.get("_metadata")
+    return metadata
+
+
+def is_home_page(mapped_har):
+    metadata = mapped_har.get("metadata")
+    if metadata and "crawl_depth" in metadata:
+        return metadata.get("crawl_depth") == 0
+        # Only home pages have a crawl depth of 0.
+    else:
+        return True
+        # legacy default
+
+
+def partition_step(har, num_partitions):
+    """Returns a partition number based on the hashed HAR page URL"""
+
+    if not har:
+        logging.warning("Unable to partition step, null HAR.")
+        return 0
+
+    page_url = get_page_url(har)
+
+    if not page_url:
+        logging.warning("Skipping HAR: unable to get page URL (see preceding warning).")
+        return 0
+
+    _hash = hash_url(page_url)
+
+    # shift partitions by one so the zero-th contains errors
+    offset = 1
+
+    return (_hash % (num_partitions - 1)) + offset
+
+
+def get_requests(har):
+    """Parses the requests from a HAR object."""
+
+    if not har:
+        return None
+
+    page_url = get_page_url(har)
+
+    if not page_url:
+        # The page_url field indirectly depends on the get_page function.
+        # If the page data is unavailable for whatever reason, skip its requests.
+        logging.warning(
+            "Skipping requests payload: unable to get page URL (see preceding warning)."
+        )
+        return None
+
+    entries = har.get("log").get("entries")
+
+    requests = []
+
+    for request in entries:
+
+        request_url = request.get("_full_url")
```

Comment:
Same suggestion from the `all` pipeline
```suggestion
        request_url = request.get("_full_url")
        
        if not request_url:
            logging.warning('Skipping empty request URL for "%s"', page_url)
            continue
```